src/ResourceManager/Network/Commands.Network/Cortex/VpnSite/UpdateAzureRmVpnSiteCommand.cs

```diff
@@ -0,0 +1,411 @@
+
+        [Parameter(
+            ParameterSetName = CortexParameterSetNames.ByVpnSiteResourceId + CortexParameterSetNames.ByVirtualWanName,
+            Mandatory = true,
+            HelpMessage = "The name of the VirtualWan this VpnSite needs to be connected to.")]
+        [Parameter(
+            ParameterSetName = CortexParameterSetNames.ByVpnSiteObject + CortexParameterSetNames.ByVirtualWanName,
+            Mandatory = true,
+            HelpMessage = "The name of the VirtualWan this VpnSite needs to be connected to.")]
+        [Parameter(
+            ParameterSetName = CortexParameterSetNames.ByVpnSiteName + CortexParameterSetNames.ByVirtualWanName,
+            Mandatory = true,
+            HelpMessage = "The name of the VirtualWan this VpnSite needs to be connected to.")]
+        [Parameter(
+            ParameterSetName = CortexParameterSetNames.ByVpnSiteResourceId + "NoVirtualWanUpdate",
+            Mandatory = false,
+            HelpMessage = "The name of the VirtualWan this VpnSite needs to be connected to.")]
+        [Parameter(
+            ParameterSetName = CortexParameterSetNames.ByVpnSiteObject + "NoVirtualWanUpdate",
+            Mandatory = false,
+            HelpMessage = "The name of the VirtualWan this VpnSite needs to be connected to.")]
+        [Parameter(
+            ParameterSetName = CortexParameterSetNames.ByVpnSiteName + "NoVirtualWanUpdate",
+            Mandatory = false,
+            HelpMessage = "The name of the VirtualWan this VpnSite needs to be connected to.")]
+        public string VirtualWanName { get; set; }
+
+        [Parameter(
+            Mandatory = true,
+            ParameterSetName = CortexParameterSetNames.ByVpnSiteName + CortexParameterSetNames.ByVirtualWanObject,
+            HelpMessage = "The VirtualWan this VpnSite needs to be connected to.")]
+        [Parameter(
+            Mandatory = true,
+            ParameterSetName = CortexParameterSetNames.ByVpnSiteObject + CortexParameterSetNames.ByVirtualWanObject,
+            HelpMessage = "The VirtualWan this VpnSite needs to be connected to.")]
+        [Parameter(
+            Mandatory = true,
+            ParameterSetName = CortexParameterSetNames.ByVpnSiteResourceId + CortexParameterSetNames.ByVirtualWanObject,
+            HelpMessage = "The VirtualWan this VpnSite needs to be connected to.")]
+        [Parameter(
+            Mandatory = false,
+            ParameterSetName = CortexParameterSetNames.ByVpnSiteName + "NoVirtualWanUpdate",
+            HelpMessage = "The VirtualWan this VpnSite needs to be connected to.")]
+        [Parameter(
+            Mandatory = false,
+            ParameterSetName = CortexParameterSetNames.ByVpnSiteObject + "NoVirtualWanUpdate",
+            HelpMessage = "The VirtualWan this VpnSite needs to be connected to.")]
+        [Parameter(
+            Mandatory = false,
+            ParameterSetName = CortexParameterSetNames.ByVpnSiteResourceId + "NoVirtualWanUpdate",
```

Comment:
Confused about this.  Why do you have a separate parameter set for NoVirtualWanUpdate when it is already optional in all other parameter sets? And why are all the VirtualWan related parameters included in this parameter set?

Reply:
So users can select not to update the VirtualWan at all when updating the VpnSite. Not having that was causing the cmdlet to fail with ambiguous parameter set.

Reply:
The right way to do this is remove "NoVirtualWanUpdate" from all the parameters related to VirtualWan - currently parameter sets don't make sense.